### PR TITLE
[MI-2266] Add test cases for api.go

### PR DESCRIPTION
**Pull request by ayusht2810**

### CODE REVIEW

File: server/plugin/api_test.go [hunk 1]
```diff
@@ -427,6 +427,561 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+				assert.Contains(resp.Message, test.ExpectedErrorMessage)
+			}
+		})
+	}
+}
+
+func TestGetCommentsForRecord(t *testing.T) {
+	requestURL := fmt.Sprintf("%s%s", constants.PathPrefix, constants.PathCommentsForRecord)
+	requestURL = strings.Replace(requestURL, "{record_id:[0-9a-f]{32}}", testutils.GetServiceNowSysID(), 1)
+	requestMethod := http.MethodGet
```
Comment (Kshitij-Katiyar) [inline server/plugin/api_test.go:567]: just use http.MethodGet no need to create a new variable for this and check this every where

**Review by Kshitij-Katiyar — APPROVED** [review]

File: server/plugin/api_test.go [hunk 2]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+		SetupAPI             func(*plugintest.API) *plugintest.API
+		ExpectedStatusCode   int
+		ExpectedErrorMessage string
+	}{
+		"success": {
+			RequestBody: `{
+				"sys_id": "mockSysID",
+				"record_type": "incident"
+				}`,
+			ChannelID: "bnqnzipmnir4zkkj95ggba5pde",
```
Comment (manojmalik20) [inline server/plugin/api_test.go:407]: Use the testutil here.

File: server/plugin/api_test.go [hunk 3]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+		RequestBody          string
+		ChannelID            string
+		SetupAPI             func(*plugintest.API) *plugintest.API
+		ExpectedStatusCode   int
+		ExpectedErrorMessage string
+	}{
+		"success": {
+			RequestBody: `{
+				"sys_id": "mockSysID",
+				"record_type": "incident"
```
Comment (manojmalik20) [inline server/plugin/api_test.go:405]: Use the constant variable here.

File: server/plugin/api_test.go [hunk 4]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+				}`,
+			ChannelID: "bnqnzipmnir4zkkj95ggba5pde",
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("GetUser", mock.AnythingOfType("string")).Return(
```
Comment (manojmalik20) [inline server/plugin/api_test.go:409]:
```suggestion
				api.On("GetUser", testutils.GetID()).Return(
```

File: server/plugin/api_test.go [hunk 5]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				api.On("LogError", constants.ErrorInvalidChannelID).Return()
+				return api
+			},
+			ExpectedErrorMessage: constants.ErrorInvalidChannelID,
+			ExpectedStatusCode:   http.StatusBadRequest,
+		},
+		"invalid request body": {
+			RequestBody: "",
+			ChannelID:   "bnqnzipmnir4zkkj95ggba5pde",
```
Comment (manojmalik20) [inline server/plugin/api_test.go:431]: Use testutil everywhere wherever possible.

File: server/plugin/api_test.go [hunk 6]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+		})
+	}
+}
+
+func TestGetCommentsForRecord(t *testing.T) {
+	requestURL := fmt.Sprintf("%s%s", constants.PathPrefix, constants.PathCommentsForRecord)
+	requestURL = strings.Replace(requestURL, "{record_id:[0-9a-f]{32}}", testutils.GetServiceNowSysID(), 1)
+	for name, test := range map[string]struct {
+		RecordType           string
+		SetupAPI             func(*plugintest.API) *plugintest.API
```
Comment (manojmalik20) [inline server/plugin/api_test.go:523]:
```suggestion
		SetupAPI             func(*plugintest.API)
```
Can you make this change in the whole file? It should work just like the SetupClient function is working.

File: server/plugin/api_test.go [hunk 7]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+			SetupClient: func(client *mock_plugin.Client) {
+				client.On("GetAllComments", constants.RecordTypeIncident, testutils.GetServiceNowSysID()).Return(
+					testutils.GetServiceNowComments(), http.StatusOK, nil,
+				)
+
+				monkey.Patch(json.Marshal, func(_ interface{}) ([]byte, error) {
+					return nil, fmt.Errorf("marshal error")
+				})
+			},
+			ExpectedStatusCode: http.StatusOK,
```
Comment (manojmalik20) [inline server/plugin/api_test.go:579]: Should the status code be `200` in this case?

File: server/plugin/api_test.go [hunk 8]
```diff
@@ -374,7 +370,549 @@ func TestGetRecordFromServiceNow(t *testing.T) {
...
+		ExpectedCount        int
+		ExpectedErrorMessage string
+	}{
+		"success": {
+			RecordType: constants.RecordTypeFollowOnTask,
+			SetupAPI: func(api *plugintest.API) *plugintest.API {
+				return api
+			},
+			SetupClient: func(client *mock_plugin.Client) {
+				client.On("GetStatesFromServiceNow", constants.RecordTypeTask).Return(
```
Comment (manojmalik20) [inline server/plugin/api_test.go:724]: This record type is different from the record type on line 719. Shouldn't it cause an error?

Comment (ayusht2810) [reply server/plugin/api_test.go:694]: there is a condition present in api in which we change the record type to `RecordTypeTask` if it is `RecordTypeFollowOnTask`

**Review by manojmalik20 — APPROVED** [review]

**Pull request merged**